### ISSUE TITLE
Java OutOfMemoryError with some Sequence diagrams

### ISSUE THREAD

**Issue by SlavaVedernikov** [plantuml-stdlib/C4-PlantUML#348]
I'm generating C4-PlantUML diagrams as output from my own framework, and it works great in general. thank you.

I'm having issues with some Sequence diagrams though. Java throws an OutOfMemoryError.

![image](https://github.com/plantuml-stdlib/C4-PlantUML/assets/19991188/3094a606-32bc-44c5-ae5e-4bfed3d4a97b)

Here is an example of the Sequence diagram that does not render.
```
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Sequence.puml

title Order Api - Api - Order Controller - Create Order - C4 Sequence - Component level

System_Ext(C4InterFlow.SoftwareSystems.ExternalSystem, "External", "")

System_Boundary(OrderApi, "Order Api")
    
Container_Boundary(OrderApi.Containers.Api, "Api")
    Component(OrderApi.Containers.Api.Components.OrderController, "Order Controller", "", "")
Boundary_End()
    
Container_Boundary(OrderApi.Containers.Data, "Data")
    Component(OrderApi.Containers.Data.Components.OrderDbContext, "Order Db Context", "", "")
Boundary_End()
    
Container_Boundary(OrderApi.Containers.Application, "Application")
    Component(OrderApi.Containers.Application.Components.OrderService, "Order Service", "", "")
    Component(OrderApi.Containers.Application.Components.DistributedLockProvider, "Distributed Lock Provider", "", "")
    Component(OrderApi.Containers.Application.Components.UserAuditLogService, "User Audit Log Service", "", "")
Boundary_End()
    
Container_Boundary(OrderApi.Containers.Integration, "Integration")
    Component(OrderApi.Containers.Integration.Components.InfinaProxy, "Infina Proxy", "", "")
    Component(OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "User Audit Log Api Proxy", "", "")
Boundary_End()
Boundary_End()

Rel(C4InterFlow.SoftwareSystems.ExternalSystem, OrderApi.Containers.Api.Components.OrderController, "Create Order")
group Create Order
alt request.IsInternalOrder
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Api.Components.OrderController, "Create Internal Order")
group Create Internal Order
alt request.Side == OrderSide.Buy
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Api.Components.OrderController, "Calculate Fee Amount")
end
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Data.Components.OrderDbContext, "Orders Add")
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Data.Components.OrderDbContext, "Outbox Messages Add")
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Api.Components.OrderController, "Retun (response)")
end
else 
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Application.Components.OrderService, "Create Order")
group Create Order
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.DistributedLockProvider, "Try Acquire Lock Async")
alt handle != null
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Map Channel To Infina Programme Name")
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Map Client Id To Infina Platform Name")
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Integration.Components.InfinaProxy, "Create Order")
group Create Order
Rel(OrderApi.Containers.Integration.Components.InfinaProxy, OrderApi.Containers.Integration.Components.InfinaProxy, "Retun (proxyResponse)")
end
alt apiResponse.HasError
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Throw Exception Create Order Failed")
group Throw Exception Create Order Failed
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Get Problem Details Exception By Infina Oms Error")
group Get Problem Details Exception By Infina Oms Error
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Extract Error Code From Infina Oms Error Message")
alt match.Success
alt int.TryParse(errorCodeMatch.Value, out int code)
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Retun (code)")
end
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Retun (problem)")
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Exception (ProblemDetailsException)")
end
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Send Order Created Log To User Audit Log")
group Send Order Created Log To User Audit Log
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Get Order Side")
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Get Order Type")
alt request.Type == OrderType.Limit
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.UserAuditLogService, "Send User Audit Log")
group Send User Audit Log
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Fallback Policy")
group Get Async Fallback Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Create Outbox Message")
group Create Outbox Message
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Data.Components.OrderDbContext, "Outbox Messages Add")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (fallback)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Retry Policy")
group Get Async Retry Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (retryPolicy)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Circuit Breaker Policy")
group Get Async Circuit Breaker Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (breaker)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Create User Audit Log")
group Create User Audit Log
alt responseMessage.IsSuccessStatusCode
Rel(OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Retun (proxyResponse)")
end
Rel(OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Retun (proxyResponse)")
end
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Retun")
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.UserAuditLogService, "Send User Audit Log")
group Send User Audit Log
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Fallback Policy")
group Get Async Fallback Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Create Outbox Message")
group Create Outbox Message
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Data.Components.OrderDbContext, "Outbox Messages Add")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (fallback)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Retry Policy")
group Get Async Retry Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (retryPolicy)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Get Async Circuit Breaker Policy")
group Get Async Circuit Breaker Policy
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Application.Components.UserAuditLogService, "Retun (breaker)")
end
Rel(OrderApi.Containers.Application.Components.UserAuditLogService, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Create User Audit Log")
group Create User Audit Log
alt responseMessage.IsSuccessStatusCode
Rel(OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Retun (proxyResponse)")
end
Rel(OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, OrderApi.Containers.Integration.Components.UserAuditLogApiProxy, "Retun (proxyResponse)")
end
end
end
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Retun (response)")
else 
Rel(OrderApi.Containers.Application.Components.OrderService, OrderApi.Containers.Application.Components.OrderService, "Exception (Exception)")
end
end
end
Rel(OrderApi.Containers.Api.Components.OrderController, OrderApi.Containers.Api.Components.OrderController, "Retun (Created)")
end


@enduml
```

...and here is the same diagram written just in PlantUML, which renders successfully.

```
@startuml

title Order Api - Api - Order Controller - Delete Order - Sequence - Component level

participant "External" as C4InterFlow.SoftwareSystems.ExternalSystem

box "Order Api" #White
    
box "Api" #White
    participant "Order Controller" as OrderApi.Containers.Api.Components.OrderController
end box

    
box "Application" #White
    participant "Order Service" as OrderApi.Containers.Application.Components.OrderService
    participant "Distributed Lock Provider" as OrderApi.Containers.Application.Components.DistributedLockProvider
    participant "User Audit Log Service" as OrderApi.Containers.Application.Components.UserAuditLogService
end box

    
box "Integration" #White
    participant "Infina Proxy" as OrderApi.Containers.Integration.Components.InfinaProxy
    participant "User Audit Log Api Proxy" as OrderApi.Containers.Integration.Components.UserAuditLogApiProxy
end box

    
box "Data" #White
    participant "Order Db Context" as OrderApi.Containers.Data.Components.OrderDbContext
end box

end box


C4InterFlow.SoftwareSystems.ExternalSystem -> OrderApi.Containers.Api.Components.OrderController : Delete Order
group Delete Order
OrderApi.Containers.Api.Components.OrderController -> OrderApi.Containers.Application.Components.OrderService : Delete Order
group Delete Order
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.DistributedLockProvider : Try Acquire Lock Async
alt handle != null
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Integration.Components.InfinaProxy : Get Order
group Get Order
alt responseMessage.IsSuccessStatusCode
OrderApi.Containers.Integration.Components.InfinaProxy -> OrderApi.Containers.Integration.Components.InfinaProxy : Retun (proxyResponse)
end
OrderApi.Containers.Integration.Components.InfinaProxy -> OrderApi.Containers.Integration.Components.InfinaProxy : Retun (proxyResponse)
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Throw Exception If Get Order Failed While Delete Order
group Throw Exception If Get Order Failed While Delete Order
alt !proxyResponse.HasError
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Retun
end
alt proxyResponse.ProblemDetails.Status == StatusCodes.Status404NotFound
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Exception (ProblemDetailsException)
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Get Problem Details Exception By Infina Oms Error
group Get Problem Details Exception By Infina Oms Error
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Extract Error Code From Infina Oms Error Message
alt match.Success
alt int.TryParse(errorCodeMatch.Value, out int code)
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Retun (code)
end
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Retun (problem)
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Exception (ProblemDetailsException)
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Map Client Id To Infina Platform Name
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Map Channel To Infina Programme Name
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Integration.Components.InfinaProxy : Delete Order
group Delete Order
alt responseMessage.IsSuccessStatusCode
OrderApi.Containers.Integration.Components.InfinaProxy -> OrderApi.Containers.Integration.Components.InfinaProxy : Retun (proxyResponse)
end
OrderApi.Containers.Integration.Components.InfinaProxy -> OrderApi.Containers.Integration.Components.InfinaProxy : Retun (proxyResponse)
end
alt response.HasError
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Throw Delete Order Error Exception
group Throw Delete Order Error Exception
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Get Problem Details Exception By Infina Oms Error
group Get Problem Details Exception By Infina Oms Error
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Extract Error Code From Infina Oms Error Message
alt match.Success
alt int.TryParse(errorCodeMatch.Value, out int code)
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Retun (code)
end
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Retun (problem)
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Exception (ProblemDetailsException)
end
end
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Send Order Cancelled Log To User Audit Log
group Send Order Cancelled Log To User Audit Log
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Get Order Side
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Get Order Type
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.UserAuditLogService : Send User Audit Log
group Send User Audit Log
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Get Async Fallback Policy
group Get Async Fallback Policy
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Create Outbox Message
group Create Outbox Message
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Data.Components.OrderDbContext : Outbox Messages Add
end
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Retun (fallback)
end
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Get Async Retry Policy
group Get Async Retry Policy
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Retun (retryPolicy)
end
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Get Async Circuit Breaker Policy
group Get Async Circuit Breaker Policy
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Application.Components.UserAuditLogService : Retun (breaker)
end
OrderApi.Containers.Application.Components.UserAuditLogService -> OrderApi.Containers.Integration.Components.UserAuditLogApiProxy : Create User Audit Log
group Create User Audit Log
alt responseMessage.IsSuccessStatusCode
OrderApi.Containers.Integration.Components.UserAuditLogApiProxy -> OrderApi.Containers.Integration.Components.UserAuditLogApiProxy : Retun (proxyResponse)
end
OrderApi.Containers.Integration.Components.UserAuditLogApiProxy -> OrderApi.Containers.Integration.Components.UserAuditLogApiProxy : Retun (proxyResponse)
end
end
end
else 
OrderApi.Containers.Application.Components.OrderService -> OrderApi.Containers.Application.Components.OrderService : Exception (Exception)
end
end
OrderApi.Containers.Api.Components.OrderController -> OrderApi.Containers.Api.Components.OrderController : Retun (Accepted)
end


@enduml

```

Please let me know if I'm doing something unexpected/wrong etc. ...or maybe it's a bug (?)

**Comment by kirchsth** [comment]
Hi @SlavaVedernikov 

I found the problem. It is `!pragma teoz true`. I uses the `!pragma teoz true` that the sequence diagram supports multiple (in your case container) boundaries too.

E.g. in your case with the `!pragma teoz true` (default of C4_Squenece.puml) your components/participants looks like below (incl. "Order API" boundary)

[![](https://www.plantuml.com/plantuml/png/hPDFYzim4CNl-HHgJYaqyYMdddQRRA6GqeMmvo1RirbHMN97e_pvzXqvjcmmXPGaXzZpvkdffn5-2gIGOcrdduohRDGWFeYQyBKeK9tamT17B6C0hBmZS2GhNnUDLIwjMGRIrfJ5UhLyJTBRpsrHgq20BErty2U2gq0sANv6XYo8NwW1nNDZnFBwtobhZaTlBI1BQmH5lPVBbUYpKkNhnZj64HQEmC6xIo2ezozdMgnNFvWIlrj_aZl_JYU5S6q7oNr0f-or_YBclJ3dz_dddFFDHwSLNXRjzip8xKpDFi6_MS9LnW5ExJBtWhok7jPbzd-vPNO4sLg6kMICGKha7NsFiNznUl5_a1j5YaFIuoRCP9nmRihaW3Dbp4qfEkancCzDOqsbo7Z7MQFg9kYHVyA-0poQ2Z9vLsViE_8t9X2QCXBehQz-lw8_6jtUvQWZKalqlKTi-nRO6RMXhJyCfqkgQ6N1UZxcGsudVNS7x6ydLpVUJlQFZv7aT-CK3-PyIQ5j9TeoGz-HFXuIhsXpfrFYpdZBwPmcvHCu7Mlx5m00)](https://www.plantuml.com/plantuml/uml/hPDFYzim4CNl-HHgJYaqyYMdddQRRA6GqeMmvo1RirbHMN97e_pvzXqvjcmmXPGaXzZpvkdffn5-2gIGOcrdduohRDGWFeYQyBKeK9tamT17B6C0hBmZS2GhNnUDLIwjMGRIrfJ5UhLyJTBRpsrHgq20BErty2U2gq0sANv6XYo8NwW1nNDZnFBwtobhZaTlBI1BQmH5lPVBbUYpKkNhnZj64HQEmC6xIo2ezozdMgnNFvWIlrj_aZl_JYU5S6q7oNr0f-or_YBclJ3dz_dddFFDHwSLNXRjzip8xKpDFi6_MS9LnW5ExJBtWhok7jPbzd-vPNO4sLg6kMICGKha7NsFiNznUl5_a1j5YaFIuoRCP9nmRihaW3Dbp4qfEkancCzDOqsbo7Z7MQFg9kYHVyA-0poQ2Z9vLsViE_8t9X2QCXBehQz-lw8_6jtUvQWZKalqlKTi-nRO6RMXhJyCfqkgQ6N1UZxcGsudVNS7x6ydLpVUJlQFZv7aT-CK3-PyIQ5j9TeoGz-HFXuIhsXpfrFYpdZBwPmcvHCu7Mlx5m00)

Without teoz (or in our case an additional `!pragma teoz false`) they looks like below ("Order API" boundary is lost).

[![](https://www.plantuml.com/plantuml/png/hPD1Zzem48Nl-HMpUocLIhXmwcbRs4gLgBeIsZDoae6rwjZfU0oalxxZX9X8KIMgPGyBxytppEUnUFAiY4DjiWTjInCgX3VcndzUB4YTyeFcjr04ZrGwosWvBrszQ8ooySpSSsLqiLWjvo_HUlsnMTJACv9Okot-3cXBp9lOFlkmTyQuaxO7sEipUcXT8AYnRXorq90h3DRPGqFgK2jWT7zWhup7B6FD1k4dLKZmfT4ml_pldPM0aRH64cj5g7Z8Ybp2G16LaraMBOF18mhIjXNMUlTyvjbg-TqAzpSXpBTkpoT5s9TzBdKagqolFy7ZO3pAzyUFgSzN5sobg9rruuLHoebQSY1_MSHLsY9Dutcg-Rm_VJsNsF-LpbF2vrtakfU4SRKwyfv-mDWzsshsVv1hnKgQn8-RC6DmmhakOWBFd33N1Loiks3ATXgZIyNQMUarKZT1Z_8JzYtIKPUOo2ywOR-Z_rfx9bq4nchZobylv8wwwjvoL85OWg5sZx6lNfAXqhnnX-ljeWkT3UAdQzxbTUBlxa33woHru-ka_FWQqTvhgsGnvrQQzWewcQ3VqNsy93dHZvbiIIhZaTCzJUGJsYhKvYy0)](https://www.plantuml.com/plantuml/png/hPD1Zzem48Nl-HMpUocLIhXmwcbRs4gLgBeIsZDoae6rwjZfU0oalxxZX9X8KIMgPGyBxytppEUnUFAiY4DjiWTjInCgX3VcndzUB4YTyeFcjr04ZrGwosWvBrszQ8ooySpSSsLqiLWjvo_HUlsnMTJACv9Okot-3cXBp9lOFlkmTyQuaxO7sEipUcXT8AYnRXorq90h3DRPGqFgK2jWT7zWhup7B6FD1k4dLKZmfT4ml_pldPM0aRH64cj5g7Z8Ybp2G16LaraMBOF18mhIjXNMUlTyvjbg-TqAzpSXpBTkpoT5s9TzBdKagqolFy7ZO3pAzyUFgSzN5sobg9rruuLHoebQSY1_MSHLsY9Dutcg-Rm_VJsNsF-LpbF2vrtakfU4SRKwyfv-mDWzsshsVv1hnKgQn8-RC6DmmhakOWBFd33N1Loiks3ATXgZIyNQMUarKZT1Z_8JzYtIKPUOo2ywOR-Z_rfx9bq4nchZobylv8wwwjvoL85OWg5sZx6lNfAXqhnnX-ljeWkT3UAdQzxbTUBlxa33woHru-ka_FWQqTvhgsGnvrQQzWewcQ3VqNsy93dHZvbiIIhZaTCzJUGJsYhKvYy0)

If you don't need the overall boundary then you can simply fix it with an `!pragma teoz false`  line after the include.

```puml
@startuml
!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Sequence.puml
'following fixes your memory problem
!pragma teoz false
```

[![](https://www.plantuml.com/plantuml/png/xPbVJ-Cu4C3V-oa6Nguh7U65fvEG5WhS8S4jekoz8ZUP5ciTE-i_qDodlnabJiC5KBkfI1-w3ooU6Ti_Z-tnJFZglB0-v6fq87MgGeRmv7tX_ZW-jk8bMKZ_56R1eKsDzgXzafhyk512SvyZvpCbPyUJawCx5ds_lJdEXVDeIVGunPy1TOf9mSF_DZTAcHUf5p2NItHGcc0XnzpO4WfhPWhpqK5XnI8NuD7y2tEX78v6Ndg5yCrcQE6ia71K_wma4yAoD39Q4aqi2XzjgNa2aO5R9Y-CHkr1uJEgqMXQ4cd-UBdquyd9jIRgAm9CfcRkNuJ5IkqIqgFLGbNjt-4m2WxfzyClpJZd9kXCs7BCqrjY97L3JNP0_qQCAwH6spLF6frBgjvDlnNxUzP9O-4IDh4hlpGOAn6Jr_GHu_5IP-FD82-45pG8_xSM9XjsE2zcR85Bts1UpA0MzSMCtYcKJ8MNHjDOhTR7qDtU7VOfscUPOaDUjoDsd_4lfFDMpeB7xCQaF-wiUPOPybwsDC0gYBfjJFlTaMN8fBynYzNgM0eiza3oxZBxx0xVkuMDknDRw-vEOzzU1eldKWjopBAaGQiMSBECq7r6Rpk9UbJJTBn4cZZbstxgD4Vtg3QAHByK2zenaf0MreG2siAHK1uinqtdasl7E3nffMF4pQVjXne9tcPkj5tugSmGJayh3JUIyr1-yXA4IeDYpYj4ECjDq9vMW3hhpx5nQ7LmbcKq--VE7FpCBE4MdHCBZ0YVkWdtw8E6iKL7DWw_r1k0oY7q9TdyiLZZ1ZBJBu-_bLVcmPPmbluCqcBrofovKgT4o_VhIUXC8HoSWWvAzQBjuSPRGLuZ58qA7Woielp2YZn7-5laIC13qYb9AhZEsdnA-Bcn-VRnrdyIUnp1DORlWLLVr8ARz_-xhNpmH26ZEFbBk4jhZHrgWn-Uh7c1osMA1Lk-hZskX5IODPvTmtQePVo97kwgwWikq1EBQu6Ub_7SVij9plvk5hLnpw6MI6cJ5Qcl861YCeGhQ_8ENtoZwYYN2vy-9TEGfYHc2KZj4mgCTy8w72DtuS5k1Tlz8rIWoKpmR0OfAOQAFV4QCKFprkbiO9gwn2UW8M5M1tHSdztww3QA7V3ObEQkkjUH8kEAWX-HrpL6SnNNxJ1aaAdmE0idxC4v7ig2EThL0ekQQe4RoOtCfU_FsglsvgtzSDlxx71dqlxGqSzLmWXNGgcPuAyKXhgN3Vgx5hknWFX0lwfT6lWtjTi7xrTXeSvsmfSn_i_hVOxnTSSEAb7QygrJsbNlb5ijWrLSE-hPYRHfa1xEBOeVQBi-_jXmfxmzgz2sx-aU7nvZi7hdaNXJMpryTIbL1uxasjN9xjGB7npdjd65_GXx3_1HORZxYCDdt_roxdqciyzazfdCFfFPPpBxJ6QVoUmpcU6_9TesbN88E_0viVcruwe1_jHPVmMCA5_fHyZLVm00)](https://www.plantuml.com/plantuml/uml/xPbVJ-Cu4C3V-oa6Nguh7U65fvEG5WhS8S4jekoz8ZUP5ciTE-i_qDodlnabJiC5KBkfI1-w3ooU6Ti_Z-tnJFZglB0-v6fq87MgGeRmv7tX_ZW-jk8bMKZ_56R1eKsDzgXzafhyk512SvyZvpCbPyUJawCx5ds_lJdEXVDeIVGunPy1TOf9mSF_DZTAcHUf5p2NItHGcc0XnzpO4WfhPWhpqK5XnI8NuD7y2tEX78v6Ndg5yCrcQE6ia71K_wma4yAoD39Q4aqi2XzjgNa2aO5R9Y-CHkr1uJEgqMXQ4cd-UBdquyd9jIRgAm9CfcRkNuJ5IkqIqgFLGbNjt-4m2WxfzyClpJZd9kXCs7BCqrjY97L3JNP0_qQCAwH6spLF6frBgjvDlnNxUzP9O-4IDh4hlpGOAn6Jr_GHu_5IP-FD82-45pG8_xSM9XjsE2zcR85Bts1UpA0MzSMCtYcKJ8MNHjDOhTR7qDtU7VOfscUPOaDUjoDsd_4lfFDMpeB7xCQaF-wiUPOPybwsDC0gYBfjJFlTaMN8fBynYzNgM0eiza3oxZBxx0xVkuMDknDRw-vEOzzU1eldKWjopBAaGQiMSBECq7r6Rpk9UbJJTBn4cZZbstxgD4Vtg3QAHByK2zenaf0MreG2siAHK1uinqtdasl7E3nffMF4pQVjXne9tcPkj5tugSmGJayh3JUIyr1-yXA4IeDYpYj4ECjDq9vMW3hhpx5nQ7LmbcKq--VE7FpCBE4MdHCBZ0YVkWdtw8E6iKL7DWw_r1k0oY7q9TdyiLZZ1ZBJBu-_bLVcmPPmbluCqcBrofovKgT4o_VhIUXC8HoSWWvAzQBjuSPRGLuZ58qA7Woielp2YZn7-5laIC13qYb9AhZEsdnA-Bcn-VRnrdyIUnp1DORlWLLVr8ARz_-xhNpmH26ZEFbBk4jhZHrgWn-Uh7c1osMA1Lk-hZskX5IODPvTmtQePVo97kwgwWikq1EBQu6Ub_7SVij9plvk5hLnpw6MI6cJ5Qcl861YCeGhQ_8ENtoZwYYN2vy-9TEGfYHc2KZj4mgCTy8w72DtuS5k1Tlz8rIWoKpmR0OfAOQAFV4QCKFprkbiO9gwn2UW8M5M1tHSdztww3QA7V3ObEQkkjUH8kEAWX-HrpL6SnNNxJ1aaAdmE0idxC4v7ig2EThL0ekQQe4RoOtCfU_FsglsvgtzSDlxx71dqlxGqSzLmWXNGgcPuAyKXhgN3Vgx5hknWFX0lwfT6lWtjTi7xrTXeSvsmfSn_i_hVOxnTSSEAb7QygrJsbNlb5ijWrLSE-hPYRHfa1xEBOeVQBi-_jXmfxmzgz2sx-aU7nvZi7hdaNXJMpryTIbL1uxasjN9xjGB7npdjd65_GXx3_1HORZxYCDdt_roxdqciyzazfdCFfFPPpBxJ6QVoUmpcU6_9TesbN88E_0viVcruwe1_jHPVmMCA5_fHyZLVm00)

Best regards
Helmut

PS.: I will enter a defect in forum, but I'm not sure if we can get an fix

**Comment by kirchsth** [comment]
Hi @SlavaVedernikov,

a second alternative could be that you increase the available memory, e.g. start the tool with 8GB like 
```bash
java  -Xms8192m  -Xmx8192m  -jar plantuml.jar "YOUR_FILE.puml"
```
but then the build is very slow (I checked it on my machine and the C4-Stdlib version requires ~1min40sec instead of ~5sec if you build it without teoz)

BR
Helmut

**Comment by SlavaVedernikov** [comment]
Hi @kirchsth 

Thank you for investigating this.
I think I'll stick with `!pragma teoz false` for now.

Thanks,
Slava

**Comment by Potherca** [comment]
Not sure yet, but we might might want to add "does not always play well with teoz" to the docs "somewhere".

**Comment by arnaudroques** [comment]
The root cause of this issue has been identified: https://github.com/plantuml/plantuml/pull/1777#issuecomment-2118408574

We will commit the fix in the incoming days.
Thanks!

**Comment by The-Lum** [comment]
Hi all,

FYI: This is now fixed on PlantUML [V1.2024.5](https://github.com/plantuml/plantuml/releases/tag/v1.2024.5).
- Thanks @SlavaVedernikov  for the report.
- Thanks PlantUML team and @arnaudroques for your work.

Regards.
